### PR TITLE
Add extra ppa inside chroot

### PR DESCRIPTION
**Pull request by chrisglass**
This PR adds an optional --ppa parameter allowing to inject custom package inside the build chroot.

### CODE REVIEW

File: generate_build_config.py [hunk 1]
```diff
@@ -26,30 +27,41 @@
...
 
     :param output_file:
         The path for the file to write to.
+    :param ppa:
+        An optional extra PPA to inject in the build chroot.
     """
+    
+    if ppa:
+        # If we passed a PPA, inject a line in the template.
+        ppa = "\n- chroot $CHROOT_ROOT add-apt-repository -y -u {}".format(ppa)
```
Comment (OddBloke) [inline generate_build_config.py:50]: Will this format allow us to pass in private PPAs?

File: tests.py [hunk 2]
```diff
@@ -48,5 +55,17 @@ def test_main_passes_first_argument_to_write_cloud_config(self, mocker):
...
+        output_filename = 'output.yaml'
+        mocker.patch('sys.argv', ['ubuntu-standalone-builder.py',
+            output_filename, '--ppa', 'ppa:foo/bar'])
+        write_cloud_config_mock = mocker.patch(
+            'generate_build_config._write_cloud_config')
+        generate_build_config.main()
+        assert [mocker.call(output_filename, ppa="ppa:foo/bar")] == \
+            write_cloud_config_mock.call_args_list
+
+
```
Comment (OddBloke) [inline tests.py:71]: Extra newlines?

File: generate_build_config.py [hunk 3]
```diff
@@ -26,30 +27,41 @@
...
 
 
 def main():
-    if len(sys.argv) != 2:
-        print('{} expects exactly one argument.'.format(sys.argv[0]))
-        sys.exit(1)
-    _write_cloud_config(sys.argv[1])
+    parser = argparse.ArgumentParser()
+    parser.add_argument('output_filename')
+    parser.add_argument('--ppa', dest='ppa', default="")
```
Comment (OddBloke) [inline generate_build_config.py:61]: Why default to `""` instead of `None`?

File: generate_build_config.py (hunk 1 above)
Comment (chrisglass) [reply generate_build_config.py:50]: No - I'll handle their case in another PR.

File: generate_build_config.py [hunk 4]
```diff
@@ -26,30 +27,41 @@
 
 # Perform the build
 - /home/ubuntu/launchpad-buildd/mount-chroot $BUILD_ID
-- /home/ubuntu/launchpad-buildd/update-debian-chroot $BUILD_ID
+- /home/ubuntu/launchpad-buildd/update-debian-chroot $BUILD_ID{}
```
Comment (OddBloke) [inline generate_build_config.py:30]: I think we could just have this on a newline, which would make the code below a bit neater.

```
In [5]: yaml.load("""\
   ...: x:
   ...: - foo
   ...: 
   ...: - bar""")
Out[5]: {'x': ['foo', 'bar']}
```

Comment (OddBloke) [inline generate_build_config.py:30]: Could we make this a named placeholder? It makes reading through the template a little easier.

**Review by OddBloke — CHANGES_REQUESTED** [review]

**Comment by OddBloke** [comment]
AIUI, this is being reworked, so I'm closing it out.

**Pull request closed without merging**